File: catapult/python/01_account_generate.py
from json import dumps

import httpx
from helper import get_config
from symbolchain.CryptoTypes import Hash256, PrivateKey
from symbolchain.facade.SymbolFacade import SymbolFacade
from symbolchain.symbol.Network import Network


def account_generate():
    config = get_config()
    url = config["private1"]["url"]

    network = httpx.get(url + "/network/properties").json()
    generation_hash = network["network"]["generationHashSeed"]

    facade = SymbolFacade("mijinnet")
    facade.network = Network(
        "mijinnet",
        0x60,
        Hash256(generation_hash),
    )
    accounts = dict()
    for i in range(7):
        key_pair = facade.KeyPair(PrivateKey.random())
        address = facade.network.public_key_to_address(key_pair.public_key)

        if i == 0:
            accounts["workaddress"] = {
                "address": str(address),
                "privatekey": str(key_pair.private_key),
                "mosaicId": "",
                "namespace": "",
            }
            continue

        accounts["address" + str(i)] = {
            "address": str(address),
            "privatekey": str(key_pair.private_key),
            "mosaicId": "",
            "namespace": "",
        }
    print(dumps(accounts, indent=2))


if __name__ == "__main__":
    account_generate()

File: catapult/javascript/01_account_generate.js
const { Account, RepositoryFactoryHttp } = require("@tech-bureau/symbol-sdk");
const config = require("config");
const { firstValueFrom } = require("rxjs");
const url = config.get("private1.url");

let accounts = {};

(async () => {
  const repo = new RepositoryFactoryHttp(url);

  //Get Networktype from catapult Networktypeをcatapultから取得
  const networkType = await firstValueFrom(repo.getNetworkType());
  //Get GenerationHash from catapult GenerationHashをcatapultから取得
  const ghash = await firstValueFrom(repo.getGenerationHash());

  console.log("network", networkType);
  console.log("GenerationHash", ghash);

  //送信元アカウントの作成
  const workaccount = Account.generateNewAccount(networkType);
  accounts[`workaddress`] = {
    privatekey: workaccount.privateKey,
    address: workaccount.address.plain(),
    mosaicId: "",
    namespace: "",
  };

  //送信先アカウントの作成
  for (let count = 1; count <= 6; count++) {
    const account = Account.generateNewAccount(networkType);
    accounts[`address${count}`] = {
      privatekey: account.privateKey,
      address: account.address.plain(),
      mosaicId: "",
      namespace: "",
    };
  }

  console.log(JSON.stringify(accounts, null, 2));
})();
